controllers/drcluster_controller.go

```diff
@@ -260,12 +262,83 @@ func (u *drclusterUpdater) clusterFenceHandle() error {
 	}
 
 	if u.object.Spec.ClusterFence == ramen.ClusterFenceStateFenced {
-		return fmt.Errorf("currently DRPolicy cant handle ClusterFenceStateFenced")
+		// return fmt.Errorf("currently DRPolicy cant handle ClusterFenceStateFenced")
+		return u.clusterFence()
 	}
 
 	return nil
 }
 
+func (u *drclusterUpdater) clusterFence() error {
+	// Ideally, here it should collect all the DRClusters available
+	// in the cluster and then match the appropriate peer cluster
+	// out of them by looking at the storage relationships. However,
+	// currently, DRCluster does not contain the storage relationship
+	// identity. Until that capability is not available, the alternate
+	// way is to collect all the DRPolicies and out of them choose the
+	// cluster whose region is same is current DRCluster's region.
+	// And that matching cluster is chosen as the peer cluster where
+	// the fencing resource is created to fence off this cluster.
+	drpolicies, err := util.GetAllDRPolicies(u.ctx, u.reconciler.APIReader)
+	if err != nil {
+		return fmt.Errorf("getting all drpolicies failed: %w", err)
+	}
+
+	peerCluster, err := u.getPeerCluster(drpolicies)
+	if err != nil {
+		return fmt.Errorf("failed to get the peer cluster for the cluster %s: %w",
+			u.object.Name, err)
+	}
+
+	return u.fenceCluster(peerCluster)
+}
+
+func (u *drclusterUpdater) fenceCluster(peerCluster ramen.DRCluster) error {
+	u.log.Info(fmt.Sprintf("initiating the cluster fence from the peer cluster %s", peerCluster.Name))
+	// TODO: logic to do fencing here
+	u.object.Status.Fenced = ramen.ClusterFenced
+
+	return nil
+}
+
+func (u *drclusterUpdater) getPeerCluster(list ramen.DRPolicyList) (ramen.DRCluster, error) {
+	var peerCluster ramen.DRCluster
+
+	found := false
+
+	for i := range list.Items {
+		drp := &list.Items[i]
+
+		for _, cluster := range drp.Spec.DRClusters {
+			// skip if cluster is this drCluster
+			if cluster == u.object.Name {
+				continue
+			}
+
+			drCluster := &ramen.DRCluster{}
+			if err := u.reconciler.APIReader.Get(u.ctx, types.NamespacedName{Name: cluster}, drCluster); err != nil {
```

Comment:
Do we need to hit the API server each time, or can we use the cache? IOW not use APIReader here?

Although this would be a rare operation and the cost is not high, just checking if there is a stronger reason.

Reply:
Yes, using API server may not be necessary each time. Cache can be used. Will be doing this change in a subsequent PR.